qiskit/circuit/library/n_local/efficient_su2.py

```diff
@@ -14,18 +14,121 @@
 
 from __future__ import annotations
 import typing
-from collections.abc import Callable
+from collections.abc import Callable, Iterable
 
 from numpy import pi
 
-from qiskit.circuit import QuantumCircuit
+from qiskit.circuit import QuantumCircuit, Gate
 from qiskit.circuit.library.standard_gates import RYGate, RZGate, CXGate
+from qiskit.utils.deprecation import deprecate_func
+from .n_local import n_local, BlockEntanglement
 from .two_local import TwoLocal
 
 if typing.TYPE_CHECKING:
     import qiskit  # pylint: disable=cyclic-import
 
 
+def efficient_su2(
+    num_qubits: int,
+    su2_gates: str | Gate | Iterable[str | Gate] | None = None,
+    entanglement: (
+        BlockEntanglement
+        | Iterable[BlockEntanglement]
+        | Callable[[int], BlockEntanglement | Iterable[BlockEntanglement]]
+    ) = "reverse_linear",
+    reps: int = 3,
+    skip_unentangled_qubits: bool = False,
+    skip_final_rotation_layer: bool = False,
+    parameter_prefix: str = "θ",
+    insert_barriers: bool = False,
+    name: str = "EfficientSU2",
```

Comment:
what about the parameters `inital_state, flatten` ?

Reply:
Just like for nlocal I don't these are needed for the new function, `initial_state` can be done with `compose()` and flatten is always true for this function.